lib/credentials.js

```diff
@@ -206,27 +213,67 @@ Credentials.prototype.hasTemporaryRequestKeys = function() {
   });
 };
 
+var SEP = '~';
+
+Credentials.packToJson = function(str) {
+  str = str.split(SEP).map(function(v) {
+    if (v.charAt(0) == '[' || v.charAt(0) == '{')
+      return v;
+    return '"' + v + '"';
+  }).join(',');
+  return '[' + str + ']';
+};
+
+
+Credentials.pack = function(values) {
+  if (JSON.stringify(values).indexOf(SEP) >= 0)
+    throw new Error('Cannot compress credentials containing  ' + SEP);
```

Comment:
I don't know if it's worth it but we could have a list of candidate separators e.g. `['~', '|', ... ]`. When asked to compress a credential object, we could iterate through the separator list until we find a suitable one. Afterwards, we can store the selected separator as the first char of the compressed string.